<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Patrick Cherry">
<meta name="dcterms.date" content="2023-12-05">

<title>GTEX RNA-seq Liver-specific Process and TF Identification</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/clipboard/clipboard.min.js"></script>
<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/quarto.js"></script>
<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/popper.min.js"></script>
<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/anchor.min.js"></script>
<link href="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2023_10_14-Cherry_P_technical_challenge_2_edits_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">GTEX RNA-seq Liver-specific Process and TF Identification</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Patrick Cherry </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 5, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>I use the publicly available bulk RNA-Seq data from <a href="https://www.gtexportal.org/home/downloads/adult-gtex">GTEX</a> (v8) identify biological pathways and processes specific to the liver. I also identify putative transcriptional regulators of these pathways and processes. To ensure that the analysis fits into local memory, I restrict the features in scope to be protein-coding genes, and the tissues in scope to be the following following: heart, kidney, liver, lung, muscle, pancreas, spleen, stomach, pituitary gland, and thyroid.</p>
<section id="notes-on-approach" class="level2">
<h2 class="anchored" data-anchor-id="notes-on-approach">Notes on approach</h2>
<p><a href="https://www.science.org/doi/full/10.1126/science.aaz1776">The GTEx Consortium atlas of genetic regulatory effects across human tissues</a> The Genotype-Tissue Expression (GTEx) project was established to characterize genetic effects on the transcriptome across human tissues and to link these regulatory mechanisms to trait and disease associations.</p>
<p>I’m interpreting “specific to the liver” to mean uniquely expressed in the liver compared to each other tissue in scope. I will express this in the analysis by running instructing DEseq to run a one-to-many test by ranking liver first among the tissue factors, and thenrunning a p-value metaanalysis on the p-value results for features (protein-coding genes) in all comparisons. (Meaning, it only takes one liver-to-tissue comparison revealing that a gene is NOT up-regulated in the liver to remove that gene from consideration.)</p>
</section>
<section id="patricks-hypothesis" class="level2">
<h2 class="anchored" data-anchor-id="patricks-hypothesis">Patrick’s hypothesis</h2>
<p>Given this is liver RNA-seq, I expect to see genes implicated in the liver main roles, namely:</p>
<ul>
<li>conditioning the blood by secreting important proteins into it, like serum albumin, immune complement proteins, C-reactive protein, clotting factors (pre-pro-thrombin, fibrinogen, plasminogen), etc.</li>
<li>cytochrome P450 proteins (CYPxxx), by which the liver catabolizes xenobiotics</li>
<li>glycolysis, gluconeogenesis, and fatty acid catabolism</li>
<li>bile salts catabolism and anabolism (cholesterols, hemoglobin/porphyrin)</li>
<li>amino acid conversion and catabolism</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">FALSE</span>)  <span class="co"># set default chunk option to not print the code that generated that output</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(usethis)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)                  <span class="co"># for reading and writing csvs</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fs)                     <span class="co"># for filesystem navigation in R</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)                  <span class="co"># for dataframe manipulation</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)                  <span class="co"># for dataframe group nesting and manipulation</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)                  <span class="co"># for functional programming, including on nested dataframes</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)                 <span class="co"># for reading excel files</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)                <span class="co"># for efficient string manipulation</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)               <span class="co"># for enhanced pipes</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)                  <span class="co"># for model manipulation</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)                <span class="co"># for factor manipulation</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)                <span class="co"># for additional plotting features</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ensembldb)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gage)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gageData)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TFEA.ChIP)              <span class="co"># analyze transcription factor enrichment in a gene set</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)           <span class="co"># speed up some parallelizeable processing</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">MulticoreParam</span>(<span class="dv">4</span>))     <span class="co"># set core parallelization to 4 CPU cores</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())           <span class="co"># set the default ggplot theme to be clear &amp; cartooney</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="metadata-sample-annotations" class="level1">
<h1>Metadata (sample annotations)</h1>
<p>Samples come from https://github.com/broadinstitute/gtex-v8/blob/master/data/GTEx_Analysis_v8_RNAseq_samples.txt, and have to be manually downloaded.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                               tissue_id   n
1                        Muscle_Skeletal 803
2                            Whole_Blood 755
3             Skin_Sun_Exposed_Lower_leg 701
4                   Adipose_Subcutaneous 663
5                          Artery_Tibial 663
6                                Thyroid 653
7                           Nerve_Tibial 619
8        Skin_Not_Sun_Exposed_Suprapubic 604
9                                   Lung 578
10                      Esophagus_Mucosa 555
11              Adipose_Visceral_Omentum 541
12                  Esophagus_Muscularis 515
13            Cells_Cultured_fibroblasts 504
14                 Breast_Mammary_Tissue 459
15                          Artery_Aorta 432
16                  Heart_Left_Ventricle 432
17                Heart_Atrial_Appendage 429
18                      Colon_Transverse 406
19   Esophagus_Gastroesophageal_Junction 375
20                         Colon_Sigmoid 373
21                                Testis 361
22                               Stomach 359
23                              Pancreas 328
24                             Pituitary 283
25                         Adrenal_Gland 258
26                          Brain_Cortex 255
27           Brain_Caudate_basal_ganglia 246
28 Brain_Nucleus_accumbens_basal_ganglia 246
29                              Prostate 245
30                      Brain_Cerebellum 241
31                                Spleen 241
32                       Artery_Coronary 240
33                                 Liver 226
34           Brain_Cerebellar_Hemisphere 215
35              Brain_Frontal_Cortex_BA9 209
36           Brain_Putamen_basal_ganglia 205
37                    Brain_Hypothalamus 202
38                     Brain_Hippocampus 197
39        Small_Intestine_Terminal_Ileum 187
40                                 Ovary 180
41  Brain_Anterior_cingulate_cortex_BA24 176
42     Cells_EBV-transformed_lymphocytes 174
43                  Minor_Salivary_Gland 162
44        Brain_Spinal_cord_cervical_c-1 159
45                                Vagina 156
46                        Brain_Amygdala 152
47                                Uterus 142
48                Brain_Substantia_nigra 139
49                         Kidney_Cortex  85
50                               Bladder  21
51                     Cervix_Endocervix  10
52                     Cervix_Ectocervix   9
53                        Fallopian_Tube   9
54                        Kidney_Medulla   4</code></pre>
</div>
</div>
<p>Put <em>liver</em> first in this list, because after filtering, I will use it to set the order of the factors.</p>
<p>Double-check I got all the names right</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<p>Got ’em!</p>
<section id="filter-sample-table-to-our-scoped-tissue-types" class="level2">
<h2 class="anchored" data-anchor-id="filter-sample-table-to-our-scoped-tissue-types">Filter sample table to our scoped tissue types</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>     n
1 4421</code></pre>
</div>
</div>
<p>4421 / 17382 = .254 . We are keeping ~ 25% of samples that were in GTEx v8.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                tissue_id   n percent
1         Muscle_Skeletal 803   18.00
2                 Thyroid 653   15.00
3                    Lung 578   13.00
4    Heart_Left_Ventricle 432    9.80
5  Heart_Atrial_Appendage 429    9.70
6                 Stomach 359    8.10
7                Pancreas 328    7.40
8               Pituitary 283    6.40
9                  Spleen 241    5.50
10                  Liver 226    5.10
11          Kidney_Cortex  85    1.90
12         Kidney_Medulla   4    0.09</code></pre>
</div>
</div>
</section>
<section id="import-the-related-bulk-rnaseq-counts" class="level2">
<h2 class="anchored" data-anchor-id="import-the-related-bulk-rnaseq-counts">Import the related bulk RNAseq counts</h2>
<p>Bulk RNAseq gct files came from: https://www.gtexportal.org/home/downloads/adult-gtex</p>
<p>I’ll use read counts because that’s what DEseq2’s statistical modeling requires. Unzip it first: <!-- ```{bash eval = FALSE, include = TRUE} --> <!-- # cd .. --> <!-- # gunzip *.gz --> <!-- ``` --></p>
<section id="troubleshoot-filtering-samplesmatrix-columns" class="level3">
<h3 class="anchored" data-anchor-id="troubleshoot-filtering-samplesmatrix-columns">Troubleshoot filtering samples/matrix columns</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4421</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] sample_id    donor_id     has_genotype tissue_id   
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>0 is the desired outcome of an anti-join.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 56200  4421</code></pre>
</div>
</div>
<p>Great: filtering went correctly. The dashes-to-periods issue was irksome. We have y = 4421 samples, and 56200 features.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<p>Great, we are good to continue to DEseq2 DESeqDataSet</p>
</section>
</section>
<section id="filter-features-to-protein-coding-genes" class="level2">
<h2 class="anchored" data-anchor-id="filter-features-to-protein-coding-genes">Filter features to protein-coding genes</h2>
<p>please restrict your analyses to protein-coding genes and the following tissues:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"
 [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    
[11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         
[16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        
[21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      
[26] "UNIPROT"     </code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 2
   genetype           n
   &lt;chr&gt;          &lt;int&gt;
 1 &lt;NA&gt;           21144
 2 protein-coding 19198
 3 pseudo          8918
 4 ncRNA           5820
 5 snoRNA           654
 6 other            368
 7 snRNA             75
 8 rRNA              20
 9 scRNA              2
10 unknown            1</code></pre>
</div>
</div>
<p>Actually apply the filter below</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 19198  4421</code></pre>
</div>
</div>
<section id="number-of-features" class="level3">
<h3 class="anchored" data-anchor-id="number-of-features">Number of features</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 56200</code></pre>
</div>
</div>
<p>19184/56200 = 34%, so we are keeping 34% of features in the counts matrix, because ENSEMBL annotates them as protein-coding. Thus we are jettisoning 66% of features as being non-protein coding. What the ENCODE db indicates is that annotated ENCODE entries are the most common, followed by pseudo, ncRNA, snoRNA, “other,” snRNA, rRNA, scRNA, and “unknown.”</p>
</section>
</section>
</section>
<section id="run-deseq2-analysis" class="level1">
<h1>Run DEseq2 analysis</h1>
<section id="prepare-data-object-for-deseq2" class="level2">
<h2 class="anchored" data-anchor-id="prepare-data-object-for-deseq2">Prepare data object for DEseq2</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>class: DESeqDataSet 
dim: 19198 4421 
metadata(1): version
assays(1): counts
rownames(19198): ENSG00000186092.4 ENSG00000187634.11 ...
  ENSG00000198695.2 ENSG00000198727.2
rowData names(0):
colnames(4421): GTEX.1117F.0426.SM.5EGHI GTEX.1117F.0726.SM.5GIEN ...
  GTEX.ZZPU.1426.SM.5GZZ6 GTEX.ZZPU.2626.SM.5E45Y
colData names(4): sample_id donor_id has_genotype tissue_id</code></pre>
</div>
</div>
<section id="filter-for-minimum-counts" class="level3">
<h3 class="anchored" data-anchor-id="filter-for-minimum-counts">Filter for minimum counts</h3>
<p>Filter for features (genes) where there are less than 113 or more samples with normalized counts greater than or equal to 5. (113 is half the number of liver samples in the data, allowing for at least some dispersion in at least among the samples of the tissue type we want to understand.)</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18023  4421</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 22813</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.4059253</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.06104045</code></pre>
</div>
</div>
<p>Originally, when filtering for features with a minimum number of counts over the minimum number of genes, I filtered out 22,813 (40.6%) based on low counts / coverage. <em>However, now the I’m filtering genes for counts <em>after</em> filtering for protein coding genes</em>, only 6.1% of features are dropped. This difference in proportionality indicates that low count features and protein coding genes are not necessarily independent. In fact, the non-protein-coding features were enriched for low expression levels across many of the samples.</p>
</section>
</section>
<section id="run-deseq2-analysis-1" class="level2">
<h2 class="anchored" data-anchor-id="run-deseq2-analysis-1">Run DEseq2 analysis</h2>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 237 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "tissue_id_Heart_Left_Ventricle_vs_Liver"  
 [2] "tissue_id_Heart_Atrial_Appendage_vs_Liver"
 [3] "tissue_id_Kidney_Cortex_vs_Liver"         
 [4] "tissue_id_Kidney_Medulla_vs_Liver"        
 [5] "tissue_id_Lung_vs_Liver"                  
 [6] "tissue_id_Muscle_Skeletal_vs_Liver"       
 [7] "tissue_id_Pancreas_vs_Liver"              
 [8] "tissue_id_Spleen_vs_Liver"                
 [9] "tissue_id_Stomach_vs_Liver"               
[10] "tissue_id_Pituitary_vs_Liver"             
[11] "tissue_id_Thyroid_vs_Liver"               </code></pre>
</div>
</div>
<section id="extract-all-contrast-data-from-deseq" class="level3">
<h3 class="anchored" data-anchor-id="extract-all-contrast-data-from-deseq">Extract all contrast data from DEseq</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                          contrast     n
1  Heart_Atrial_Appendage_vs_Liver 18023
2    Heart_Left_Ventricle_vs_Liver 18023
3           Kidney_Cortex_vs_Liver 18023
4          Kidney_Medulla_vs_Liver 18023
5                    Lung_vs_Liver 18023
6         Muscle_Skeletal_vs_Liver 18023
7                Pancreas_vs_Liver 18023
8               Pituitary_vs_Liver 18023
9                  Spleen_vs_Liver 18023
10                Stomach_vs_Liver 18023
11                Thyroid_vs_Liver 18023</code></pre>
</div>
</div>
</section>
</section>
<section id="prepare-data-with-additional-annotations" class="level2">
<h2 class="anchored" data-anchor-id="prepare-data-with-additional-annotations">Prepare data with additional annotations</h2>
<section id="check-if-ensg-string-manipulation-prodcuces-unique-values" class="level4">
<h4 class="anchored" data-anchor-id="check-if-ensg-string-manipulation-prodcuces-unique-values">Check if ENSG string manipulation prodcuces unique values</h4>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 2
      n     nn
  &lt;int&gt;  &lt;int&gt;
1     1 198253</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 2
      n     nn
  &lt;int&gt;  &lt;int&gt;
1     1 198253</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
</div>
</section>
<section id="further-explore-the-contrasts" class="level3">
<h3 class="anchored" data-anchor-id="further-explore-the-contrasts">Further explore the contrasts</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 110 × 4
# Groups:   contrast [11]
   contrast                        symbol  name                   log2FoldChange
   &lt;chr&gt;                           &lt;chr&gt;   &lt;chr&gt;                           &lt;dbl&gt;
 1 Heart_Atrial_Appendage_vs_Liver SLC27A5 solute carrier family…           7.77
 2 Heart_Atrial_Appendage_vs_Liver RDH16   retinol dehydrogenase…           9.76
 3 Heart_Atrial_Appendage_vs_Liver SLC22A1 solute carrier family…          11.8 
 4 Heart_Atrial_Appendage_vs_Liver CYP2E1  cytochrome P450 famil…          11.5 
 5 Heart_Atrial_Appendage_vs_Liver ASGR1   asialoglycoprotein re…          10.1 
 6 Heart_Atrial_Appendage_vs_Liver CYP2D6  cytochrome P450 famil…           8.74
 7 Heart_Atrial_Appendage_vs_Liver CPS1    carbamoyl-phosphate s…           8.98
 8 Heart_Atrial_Appendage_vs_Liver APOM    apolipoprotein M                 7.20
 9 Heart_Atrial_Appendage_vs_Liver ATF5    activating transcript…           7.22
10 Heart_Atrial_Appendage_vs_Liver MST1    macrophage stimulatin…           7.25
# ℹ 100 more rows</code></pre>
</div>
</div>
<p>This is looking promising. When sorted by statistic and the top ten genes are displayed from each tissue, there are many common genes that are relatively up-regulated (in mRNA expression) in liver compared to each of the other tissues. SLC27A5 is very commonly (and very strongly) up-regulated in liver; RDH16 and CPS1 are as well.</p>
</section>
<section id="ma-plot-analysis-qc" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-analysis-qc">MA Plot (analysis QC)</h3>
<div class="cell">
<div class="cell-output-display">
<p><img src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>For all contrasts, the data points (gene IDs) are roughly symmetric around the x-axis (indicating no systematic bias), and the majority are close to the x-axis, indicating many genes are not differentially expressed.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 × 4
   contrast             `mean log2FoldChange` `S.D. log2FoldChange` `n features`
   &lt;chr&gt;                                &lt;dbl&gt;                 &lt;dbl&gt;        &lt;int&gt;
 1 Heart_Atrial_Append…               -0.0807                  2.49        18023
 2 Heart_Left_Ventricl…               -0.160                   2.53        18023
 3 Kidney_Cortex_vs_Li…                0.120                   2.10        18023
 4 Kidney_Medulla_vs_L…               -0.0501                  2.29        18023
 5 Lung_vs_Liver                       0.134                   2.37        18023
 6 Muscle_Skeletal_vs_…               -0.342                   2.77        18023
 7 Pancreas_vs_Liver                  -0.0455                  2.21        18023
 8 Pituitary_vs_Liver                  0.229                   2.69        18023
 9 Spleen_vs_Liver                    -0.156                   2.36        18023
10 Stomach_vs_Liver                    0.112                   2.27        18023
11 Thyroid_vs_Liver                   -0.0922                  2.37        18023</code></pre>
</div>
</div>
<p>As a clearer summary plot, the histograms for each contrast a preponderance of genes are close to the 0 on the x-axis, indicating many features have limited fold change between samples.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>While the test statistic outputted by DEseq2 is less “clipped” to zero than the p-values (many of which exhaust the range and get reported as 0.000000e+00), the statistic range/distribution is not the same for all contrasts, notably for the Kidney Medulla vs Liver comparison. This can be explained from the <a href="#Filter-sample-table-to-our-scoped-tissue-types">Filter sample table to our scoped tissue types</a> section of this document, which shows the Kidney Medulla GTEx tissue set as having far fewer samples (n = 4) than the other tissue types (mean = 368, sd = 230).</p>
<p>As long as we <code>group_by()</code> when ranking, sorting, and otherwise comparing (in long format), this should be okay. Once pivoted into wide format, this can present a problem because not every column of values will be able to react similarly to the same ranking, sorting, or cut-offs.</p>
</section>
</section>
</section>
<section id="cross-contrast-analysis" class="level1">
<h1>Cross-contrast analysis</h1>
<section id="summarize-by-feature" class="level2">
<h2 class="anchored" data-anchor-id="summarize-by-feature">Summarize by feature</h2>
<p>Which is effectively a wide format, appropriate for joining. (Grouped-by calculations in the same column run more efficiently and are easier to program than row-wise multi-column calculations.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 4798 warnings in `summarize()`.
The first warning was:
ℹ In argument: `across(...)`.
ℹ In group 7: `ENSG = "ENSG00000188290.10"`, `short_ensg = "ENSG00000188290"`,
  `symbol = "HES4"`, `entrez = "57801"`, `name = "hes family bHLH transcription
  factor 4"`.
Caused by warning in `min()`:
! no non-missing arguments to min; returning Inf
ℹ Run `dplyr::last_dplyr_warnings()` to see the 4797 remaining warnings.</code></pre>
</div>
</div>
<p>To explain why I’m contriving these summary statistics, I have to first acknowledge that downstream processing (GO analysis, TF.ChIP) expects continuous data ( <em>e.g.</em> <code>log2FoldChange</code>, <code>adj.p.val</code>) for <em>both</em> the up-regulated and down-regulated features. Thus, in drawing conclusions from the comparisons, I am balancing maintaining continuous variables that can be used later with summary statistics that <em>suggest</em> consistency across all the tissue type comparisons made (and in the case of the highly confident features/genes, these suggestive consistency metrics indicate high levels of agreement).</p>
<p>These metrics are the <code>mean_log2FoldChange</code>, <code>sd_log2FoldChange</code>, and the <code>below_zero_min_log2FoldChange</code>, (as well as these versions on <code>stat</code> and <code>padj</code>). Together, these continuous feature summary metrics describe how consistently the metrics are across comparisons (e.g.&nbsp;liver to thyroid, liver to lung, etc.). When a feature is specific to the liver, the mean will be negative and large, the sd will be smaller than the mean, and the min_below_zero will not be far away from the mean. Such an outcome indicates that all the comparisons agree that a gene is specifically up-regulated in the liver compared to all other tissues considered here.</p>
<p>Applying these summary transformations to the <code>log2FoldChange</code>, <code>pvalue</code>, and <code>padj</code> are reasonable because these metrics either do not depend on the number of samples (<code>log2FoldChange</code>), or account for the number of samples in the condition (<code>pvalue</code>, and <code>padj</code>), explicitly allowing for comparisons between conditions.</p>
</section>
<section id="pivot-to-wide" class="level2">
<h2 class="anchored" data-anchor-id="pivot-to-wide">Pivot to wide</h2>
</section>
</section>
<section id="pathway-gene-ontology-go-enrichment-analysis" class="level1">
<h1>Pathway / Gene Ontology (GO) Enrichment Analysis</h1>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 5
   mean_log2FoldChange sd_log2FoldChange mean_pvalue symbol  name               
                 &lt;dbl&gt;             &lt;dbl&gt;       &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;              
 1               -7.60             0.560   1.66e-214 SLC27A5 solute carrier fam…
 2               -9.37             0.785   5.51e-156 RDH16   retinol dehydrogen…
 3               -9.48             0.401   5.40e-152 CPS1    carbamoyl-phosphat…
 4               -8.15             1.04    1.47e-147 A1BG    alpha-1-B glycopro…
 5               -9.56             1.28    1.90e-135 ASGR1   asialoglycoprotein…
 6              -10.5              0.842   1.91e-124 CYP2E1  cytochrome P450 fa…
 7              -10.5              1.93    7.35e-119 VTN     vitronectin        
 8              -10.3              0.948   7.10e-118 SLC22A1 solute carrier fam…
 9               -6.96             0.329   1.60e-116 ATF5    activating transcr…
10               -6.88             1.01    2.56e-106 ANG     angiogenin         
11              -11.3              2.46    7.90e-100 APOB    apolipoprotein B   
12              -11.5              1.01    5.25e- 98 HPX     hemopexin          
13               -7.68             1.31    2.42e- 97 ITIH4   inter-alpha-trypsi…
14              -12.0              1.89    2.99e- 92 APOA1   apolipoprotein A1  
15              -10.3              1.84    1.37e- 90 TFR2    transferrin recept…
16               -4.45             0.542   1.93e- 88 DHODH   dihydroorotate deh…
17               -9.34             1.81    6.04e- 88 APOC1   apolipoprotein C1  
18              -10.7              1.03    3.93e- 86 TF      transferrin        
19               -7.63             1.10    5.78e- 85 CYP2D6  cytochrome P450 fa…
20                4.33             2.05    6.70e- 84 AIF1L   allograft inflamma…</code></pre>
</div>
</div>
<p>Checking in on genes and hypotheses:</p>
<ol type="1">
<li>conditioning the blood by secreting important proteins into it, like serum albumin, immune complement proteins, C-reactive protein, clotting factors (pre-pro-thrombin, fibrinogen, plasminogen), etc.
<ul>
<li>fibrinogen alpha chain</li>
<li>amyloid P component, serum</li>
<li>coagulation factor IX</li>
<li>fibrinogen beta chain</li>
<li>complement C8 beta chain</li>
<li>plasminogen</li>
<li>fibrinogen gamma chain</li>
<li>C-reactive protein</li>
<li>complement factor H related 2</li>
<li>complement factor H related 5</li>
<li>complement C8 alpha chain</li>
<li>coagulation factor II, thrombin</li>
<li>albumin</li>
<li>complement C9</li>
<li>coagulation factor XIII B chain</li>
</ul></li>
<li>cytochrome P450 proteins (CYPxxx), by which the liver catabolizes xenobiotics
<ul>
<li>cytochrome P450 family 2 subfamily B member 6</li>
<li>cytochrome P450 family 2 subfamily C member 9</li>
<li>cytochrome P450 family 4 subfamily F member 2</li>
</ul></li>
<li>glycolysis, gluconeogenesis, and fatty acid catabolism
<ul>
<li>apolipoprotein C3</li>
<li>apolipoprotein H</li>
<li>apolipoprotein A2</li>
<li>glucose-6-phosphatase catalytic subunit 1</li>
<li>aldolase, fructose-bisphosphate B</li>
</ul></li>
<li>bile salts catabolism and anabolism (cholesterols, hemoglobin/porphyrin)
<ul>
<li>??? (we’ll see later in the pathways analysis)</li>
</ul></li>
<li>amino acid conversion and catabolism
<ul>
<li>alanine–glyoxylate aminotransferase</li>
<li>tyrosine aminotransferase</li>
<li>alanine–glyoxylate aminotransferase 2</li>
</ul></li>
</ol>
<p>I didn’t find many bile salts and cholesterol metabolism genes. Curious. Maybe this is a lesser function of the liver than I imagined, or maybe tissue sampling area makes a difference.</p>
<p>I also recovered some interesting “other” proteins:</p>
<ul>
<li>GC vitamin D binding protein</li>
<li>serpin family A member 7</li>
<li>hydroxyacid oxidase 1</li>
<li>UDP glucuronosyltransferase family 2 member B10</li>
<li>kininogen 1</li>
<li>mannose binding lectin 2</li>
<li>UDP glucuronosyltransferase family 2 member B4</li>
<li>serpin family A member 11</li>
<li>secreted phosphoprotein 2</li>
<li>solute carrier family 2 member 2</li>
<li>carboxypeptidase N subunit 1</li>
<li>cAMP responsive element binding protein 3 like 3</li>
<li>insulin like growth factor binding protein 1</li>
</ul>
<p>Themes on vitamin D, UDP-yl-ation (a means of preparing xenobiotics for excretion), and regulatory / signaling proteins.</p>
<section id="pathway-analysis" class="level2">
<h2 class="anchored" data-anchor-id="pathway-analysis">Pathway analysis</h2>
<p>with GAGE ### Prepare GAGE’s needed data</p>
<section id="perpare-deseq2-result-for-gage" class="level3">
<h3 class="anchored" data-anchor-id="perpare-deseq2-result-for-gage">Perpare DEseq2 result for GAGE</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>     10998       8608       1373          1        432       1571 
 -7.601432  -9.367489  -9.478086  -8.152599  -9.564476 -10.454284 </code></pre>
</div>
</div>
</section>
<section id="run-gage" class="level3">
<h3 class="anchored" data-anchor-id="run-gage">Run GAGE</h3>
</section>
<section id="view-results" class="level3">
<h3 class="anchored" data-anchor-id="view-results">View Results</h3>
<p>Remember that “down-regulated” (i.e.&nbsp;the “$less” table) is <em>up-regulated</em> in liver, due to the DEseq contrasts.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                                                         p.geomean stat.mean
hsa00830 Retinol metabolism                           2.779302e-15 -9.570937
hsa00982 Drug metabolism - cytochrome P450            1.480851e-13 -8.552116
hsa00980 Metabolism of xenobiotics by cytochrome P450 3.742583e-12 -7.909573
hsa00140 Steroid hormone biosynthesis                 7.829862e-12 -7.983659
hsa00983 Drug metabolism - other enzymes              3.227821e-11 -7.785412
hsa04610 Complement and coagulation cascades          1.590346e-10 -7.112879
hsa00260 Glycine, serine and threonine metabolism     8.543084e-08 -6.071404
hsa00053 Ascorbate and aldarate metabolism            3.398045e-07 -6.247136
hsa04146 Peroxisome                                   3.456134e-07 -5.215038
hsa00071 Fatty acid metabolism                        8.363970e-07 -5.220379
hsa00040 Pentose and glucuronate interconversions     8.446235e-07 -5.659101
hsa00860 Porphyrin and chlorophyll metabolism         3.474540e-06 -4.994513
hsa00500 Starch and sucrose metabolism                5.047958e-06 -4.774060
hsa00380 Tryptophan metabolism                        1.180799e-05 -4.535570
hsa00280 Valine, leucine and isoleucine degradation   2.437705e-05 -4.277734
hsa03320 PPAR signaling pathway                       4.831137e-05 -4.070571
hsa04976 Bile secretion                               5.608401e-05 -4.037952
hsa04975 Fat digestion and absorption                 8.538313e-05 -4.013194
hsa00120 Primary bile acid biosynthesis               1.087369e-04 -4.465205
hsa00591 Linoleic acid metabolism                     1.465698e-04 -4.016737
                                                             p.val set.size
hsa00830 Retinol metabolism                           2.779302e-15       60
hsa00982 Drug metabolism - cytochrome P450            1.480851e-13       68
hsa00980 Metabolism of xenobiotics by cytochrome P450 3.742583e-12       66
hsa00140 Steroid hormone biosynthesis                 7.829862e-12       52
hsa00983 Drug metabolism - other enzymes              3.227821e-11       48
hsa04610 Complement and coagulation cascades          1.590346e-10       69
hsa00260 Glycine, serine and threonine metabolism     8.543084e-08       32
hsa00053 Ascorbate and aldarate metabolism            3.398045e-07       22
hsa04146 Peroxisome                                   3.456134e-07       78
hsa00071 Fatty acid metabolism                        8.363970e-07       43
hsa00040 Pentose and glucuronate interconversions     8.446235e-07       28
hsa00860 Porphyrin and chlorophyll metabolism         3.474540e-06       38
hsa00500 Starch and sucrose metabolism                5.047958e-06       50
hsa00380 Tryptophan metabolism                        1.180799e-05       42
hsa00280 Valine, leucine and isoleucine degradation   2.437705e-05       44
hsa03320 PPAR signaling pathway                       4.831137e-05       70
hsa04976 Bile secretion                               5.608401e-05       70
hsa04975 Fat digestion and absorption                 8.538313e-05       44
hsa00120 Primary bile acid biosynthesis               1.087369e-04       16
hsa00591 Linoleic acid metabolism                     1.465698e-04       28</code></pre>
</div>
</div>
<p>What’s <em>down-regulated in liver</em> <em>compared to other tissues in the scope of our analysis</em>?</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                                                            p.geomean stat.mean
hsa04260 Cardiac muscle contraction                      4.226923e-05 4.0485435
hsa04510 Focal adhesion                                  8.259292e-05 3.8042688
hsa04020 Calcium signaling pathway                       4.162904e-04 3.3711847
hsa04971 Gastric acid secretion                          4.992403e-04 3.3629267
hsa04540 Gap junction                                    2.260177e-03 2.8795191
hsa04512 ECM-receptor interaction                        1.063049e-02 2.3255222
hsa00512 Mucin type O-Glycan biosynthesis                1.216786e-02 2.3178456
hsa04310 Wnt signaling pathway                           1.365917e-02 2.2195968
hsa04010 MAPK signaling pathway                          1.744611e-02 2.1151954
hsa04740 Olfactory transduction                          1.772474e-02 2.1215169
hsa04144 Endocytosis                                     1.777223e-02 2.1102663
hsa04360 Axon guidance                                   2.267218e-02 2.0114316
hsa04810 Regulation of actin cytoskeleton                2.364352e-02 1.9896259
hsa04070 Phosphatidylinositol signaling system           4.524492e-02 1.7041613
hsa04742 Taste transduction                              4.633911e-02 1.7021996
hsa04666 Fc gamma R-mediated phagocytosis                5.200086e-02 1.6354382
hsa00603 Glycosphingolipid biosynthesis - globo series   5.488970e-02 1.6593014
hsa04916 Melanogenesis                                   6.589814e-02 1.5133803
hsa04960 Aldosterone-regulated sodium reabsorption       7.526569e-02 1.4512416
hsa04970 Salivary secretion                              7.787444e-02 1.4259650
hsa04966 Collecting duct acid secretion                  9.503669e-02 1.3287755
hsa04720 Long-term potentiation                          1.029612e-01 1.2717035
hsa04722 Neurotrophin signaling pathway                  1.100257e-01 1.2305044
hsa04012 ErbB signaling pathway                          1.306016e-01 1.1275055
hsa00604 Glycosphingolipid biosynthesis - ganglio series 1.335755e-01 1.1341217
hsa04730 Long-term depression                            1.385772e-01 1.0911743
hsa04330 Notch signaling pathway                         1.424131e-01 1.0774613
hsa04962 Vasopressin-regulated water reabsorption        1.465997e-01 1.0579484
hsa03010 Ribosome                                        1.579659e-01 1.0084071
hsa00562 Inositol phosphate metabolism                   1.619384e-01 0.9910218
                                                                p.val set.size
hsa04260 Cardiac muscle contraction                      4.226923e-05       76
hsa04510 Focal adhesion                                  8.259292e-05      200
hsa04020 Calcium signaling pathway                       4.162904e-04      176
hsa04971 Gastric acid secretion                          4.992403e-04       73
hsa04540 Gap junction                                    2.260177e-03       88
hsa04512 ECM-receptor interaction                        1.063049e-02       84
hsa00512 Mucin type O-Glycan biosynthesis                1.216786e-02       30
hsa04310 Wnt signaling pathway                           1.365917e-02      148
hsa04010 MAPK signaling pathway                          1.744611e-02      262
hsa04740 Olfactory transduction                          1.772474e-02       97
hsa04144 Endocytosis                                     1.777223e-02      201
hsa04360 Axon guidance                                   2.267218e-02      129
hsa04810 Regulation of actin cytoskeleton                2.364352e-02      209
hsa04070 Phosphatidylinositol signaling system           4.524492e-02       78
hsa04742 Taste transduction                              4.633911e-02       43
hsa04666 Fc gamma R-mediated phagocytosis                5.200086e-02       93
hsa00603 Glycosphingolipid biosynthesis - globo series   5.488970e-02       14
hsa04916 Melanogenesis                                   6.589814e-02       99
hsa04960 Aldosterone-regulated sodium reabsorption       7.526569e-02       42
hsa04970 Salivary secretion                              7.787444e-02       85
hsa04966 Collecting duct acid secretion                  9.503669e-02       27
hsa04720 Long-term potentiation                          1.029612e-01       69
hsa04722 Neurotrophin signaling pathway                  1.100257e-01      127
hsa04012 ErbB signaling pathway                          1.306016e-01       87
hsa00604 Glycosphingolipid biosynthesis - ganglio series 1.335755e-01       15
hsa04730 Long-term depression                            1.385772e-01       68
hsa04330 Notch signaling pathway                         1.424131e-01       47
hsa04962 Vasopressin-regulated water reabsorption        1.465997e-01       43
hsa03010 Ribosome                                        1.579659e-01       87
hsa00562 Inositol phosphate metabolism                   1.619384e-01       57</code></pre>
</div>
</div>
<p>Note that the p-values for pathways down-regulated in the liver are less significant than those of pathways up-regulated, likely due to by high dispersion among the comparison tissues.</p>
<section id="pathway-plots" class="level4">
<h4 class="anchored" data-anchor-id="pathway-plots">Pathway plots</h4>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "hsa00830 Retinol metabolism"                          
[2] "hsa00982 Drug metabolism - cytochrome P450"           
[3] "hsa00980 Metabolism of xenobiotics by cytochrome P450"
[4] "hsa00140 Steroid hormone biosynthesis"                
[5] "hsa00983 Drug metabolism - other enzymes"             
[6] "hsa04976 Bile secretion"                              
[7] "hsa00860 Porphyrin and chlorophyll metabolism"        
[8] "hsa00280 Valine, leucine and isoleucine degradation"  </code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00830.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00982.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00980.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00140.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00983.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa04976.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00860.pathview.png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Working in directory /Users/patrick/GTEX-liver-parent-dir/GTEX-RNA-seq-liver</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Info: Writing image file hsa00280.pathview.png</code></pre>
</div>
</div>
<p>Typically, expression heat maps use the color-blind-unfriendly and unintuitive red for up-regulated expression, and green for decreased expression.</p>
<p>However, here, because I never inverted (multiplied by -1) the log2-fold-change values, green is for increased relative expression, and red is for decreased relative expression.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00830.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00830 Retinol metabolism</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00982.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00982 Drug metabolism - cytochrome P450</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00980.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00980 Metabolism of xenobiotics by cytochrome P450</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00983.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00983 Drug metabolism - other enzymes</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00140.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00140 Steroid hormone biosynthesis</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00983.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00983 Drug metabolism - other enzymes</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa04976.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa04976 Bile secretion</figcaption>
</figure>
</div>
<p>The bile secretion graphic is particularly interesting because it depicts spatial regulation across cell types, and those expectations (hypotheses) are upheld by the bulk RNA-seq data from GTEx. Most hepatocyte genes are up-regulated, whereas most cholangiocyte (epithelial cells of the bile duct) genes are not up-regulated.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="hsa00860.pathview.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">hsa00860 Porphyrin and chlorophyll metabolism</figcaption>
</figure>
</div>
<p>Porphyrin metabolism is a curious case: not many of the GO term’s genes are up-regulated, and even along a linear pathway, not all the genes in are up-regulated. I am especially surprised to see biliverdin to bilirubin conversion down-regulated relative to the other tissues in this set. Meanwhile, two of bilirubin’s downstream steps are up-regulated.</p>
</section>
</section>
</section>
</section>
<section id="transcription-factor-enrichment-analysis" class="level1">
<h1>Transcription Factor Enrichment Analysis</h1>
<p>With <code>TFEA_ChIP</code>, we will leverage the contingency matrices crossing DEseq2 up v. down and previously-existing ChIP-seq (up v. down) data sets.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Some genes returned 1:many mapping to ENTREZ ID. Genes were assigned the first ENTREZ ID match found.</code></pre>
</div>
</div>
<section id="add-chip-data" class="level2">
<h2 class="anchored" data-anchor-id="add-chip-data">Add ChIP data</h2>
<div class="cell">
<div class="cell-output-display">
<p><img src="2023_10_14-Cherry_P_technical_challenge_2_edits_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The above plot shows GSEA statistically-significant points (p ≥ .05) supported by the DEseq data, where enrichment score is a [-1, 1] normalized range, and ChIP-seq rank is ranked dataset’s ranking of correspondence with the DEseq data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>             TF       ES arg.ES pVal numberOfChIPs
FOXA2     FOXA2  0.99125      8 0.00             2
NR2F2     NR2F2  0.94015     44 0.00             3
TBP         TBP -0.87588    608 0.00             4
ELF1       ELF1 -0.90544    627 0.00             5
HNF4G     HNF4G  0.97522     19 0.01             2
FOXA1     FOXA1  0.94599     40 0.01             3
HNF4A     HNF4A  0.96642     26 0.01             3
GABPA     GABPA -0.69893    593 0.01             9
CHD2       CHD2 -0.86508    618 0.01             5
SMAD5     SMAD5 -0.99854    686 0.01             1
NR2F6     NR2F6  0.97234     20 0.03             1
SIN3A     SIN3A -0.81379    654 0.03             5
RCOR1     RCOR1 -0.82070    610 0.03             4
MXI1       MXI1 -0.78348    564 0.03             5
HNF1A     HNF1A  0.97817     16 0.04             1
FOXP1     FOXP1  0.93294     48 0.04             2
CBFB       CBFB -0.95049    660 0.04             2
MAZ         MAZ -0.88623    639 0.04             3
CREB1     CREB1 -0.80579    628 0.05             5
PML         PML -0.96507    663 0.05             1
PAX5       PAX5 -0.91108    625 0.06             2
NR5A1     NR5A1  0.95779     30 0.07             1
HCFC1     HCFC1 -0.83570    669 0.07             3
KLF5       KLF5 -0.94323    648 0.07             1
ATF3       ATF3  0.76196     11 0.08             6
PAXIP1   PAXIP1  0.93886     43 0.08             1
PBX3       PBX3 -0.97234    668 0.08             1
GFI1       GFI1  0.95342     33 0.09             1
GTF2F1   GTF2F1 -0.78511    554 0.09             4
ONECUT1 ONECUT1  0.94905     36 0.10             1
MBD4       MBD4  0.94614     38 0.10             1
TAF7       TAF7 -0.95779    658 0.10             1
HHEX       HHEX  0.95488     32 0.11             1
SOX13     SOX13  0.94469     39 0.11             1
RB1         RB1 -0.83897    620 0.11             2
ETS1       ETS1 -0.88563    638 0.12             2
ZBTB40   ZBTB40 -0.82977    650 0.12             2
RELA       RELA -0.84840    582 0.12             2
SP4         SP4 -0.90830    624 0.12             1
CREM       CREM -0.89989    672 0.13             2</code></pre>
</div>
</div>
<p>Notes on TF_ranking results:</p>
<ul>
<li>HNF4 (Hepatocyte Nuclear Factor 4) is a nuclear receptor protein mostly expressed in the liver, gut, kidney, and pancreatic beta cells that is critical for liver development.</li>
<li>Forkhead box protein A2 (FOXA2), also known as hepatocyte nuclear factor 3-beta (HNF-3B), is a transcription factor that plays an important role during development</li>
<li>ATF-3 (Cyclic AMP-dependent transcription factor) is an activating member of the mammalian activation transcription factor/cAMP responsive element-binding (CREB) protein family. <em>ATF-3 is induced upon physiological stress in various tissues.</em></li>
<li>Forkhead box protein A1 (FOXA1), also known as hepatocyte nuclear factor 3-alpha (HNF-3A), is a transcriptional activator for liver-specific transcripts such as albumin and transthyretin, and they also interact with chromatin as a pioneer factor.</li>
<li>JUND: a functional component of the AP1 transcription factor complex. It has been proposed to protect cells from p53-dependent senescence and apoptosis.</li>
<li>TAF1 is TFIID subunit 1, a general pol II transcriptional complex component.</li>
<li>REST: RE1-silencing transcription factor gene encodes a transcriptional repressor which represses neuronal genes in non-neuronal tissues.</li>
<li>HNF1 homeobox A (hepatocyte nuclear factor 1 homeobox A), also known as HNF1A, is ubiquitously expressed in many tissues and cell types. The protein encoded by this gene is a transcription factor that is highly expressed in the liver and is involved in the regulation of the expression of several liver-specific genes.</li>
</ul>
<p>The GSEA option was unhelpful at sorting TFs by the level of evidence to support them (all enrichment scores were 1, and all p-values were 0). Wilcoxon doesn’t make a pretty plot, but it gives us helpful sorting data.</p>
<p>Still, I’m not impressed by these results. Even with more stringent filtering (log2-fold-change, p-value), I still can’t get a list that looks specific to the liver.</p>
<section id="run-gsea-on-tfea.chip" class="level3">
<h3 class="anchored" data-anchor-id="run-gsea-on-tfea.chip">Run GSEA on TFEA.ChIP</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |                                                                      |   1%
  |                                                                            
  |=                                                                     |   1%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |==                                                                    |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |===                                                                   |   5%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=======                                                               |   9%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=======                                                               |  11%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |=========                                                             |  12%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |=========                                                             |  14%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |==========                                                            |  15%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |============                                                          |  18%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |=============                                                         |  19%
  |                                                                            
  |==============                                                        |  19%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |===============                                                       |  21%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |================                                                      |  24%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |===================                                                   |  26%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |===================                                                   |  28%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |=====================                                                 |  29%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=====================                                                 |  31%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |========================                                              |  35%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  36%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |============================                                          |  39%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |============================                                          |  41%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |==============================                                        |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |===============================                                       |  45%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |=================================                                     |  48%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |===================================                                   |  49%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |===================================                                   |  51%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=======================================                               |  55%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |========================================                              |  58%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  59%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |==========================================                            |  61%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |============================================                          |  64%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |==============================================                        |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |=================================================                     |  69%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |===================================================                   |  72%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |===================================================                   |  74%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |========================================================              |  81%
  |                                                                            
  |=========================================================             |  81%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |============================================================          |  85%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |=============================================================         |  86%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |=============================================================         |  88%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |===============================================================       |  91%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |===================================================================   |  95%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |====================================================================  |  98%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |===================================================================== |  99%
  |                                                                            
  |======================================================================|  99%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                    Accession   Cell Treatment      TF      ES p.val
1     ENCSR297GII.HNF4G.liver  liver             HNF4G 0.54053     0
2      ENCSR016BMM.TAF1.liver  liver              TAF1 0.49766     0
3     ENCSR735KEY.FOXA1.liver  liver             FOXA1 0.49263     0
4   ENCSR000EZP.SREBF1.Hep-G2 Hep-G2            SREBF1 0.48636     0
5     ENCSR310NYI.FOXA2.liver  liver             FOXA2 0.48343     0
6     ENCSR324RCI.FOXA1.liver  liver             FOXA1 0.48302     0
7     ENCSR000BQW.MBD4.Hep-G2 Hep-G2              MBD4 0.47427     0
8      ENCSR205FOW.ATF3.liver  liver              ATF3 0.47074     0
9      ENCSR480LIS.ATF3.liver  liver              ATF3 0.47024     0
10  ENCSR000EVQ.TCF7L2.Hep-G2 Hep-G2            TCF7L2 0.46749     0
11  ENCSR000EZO.SREBF2.Hep-G2 Hep-G2            SREBF2 0.45675     0
12   ENCSR516HUP.ZBTB33.liver  liver            ZBTB33 0.45420     0
13     ENCSR893QWP.REST.liver  liver              REST 0.45235     0
14    ENCSR080XEY.FOXA2.liver  liver             FOXA2 0.44996     0
15     ENCSR736BUG.EGR1.liver  liver              EGR1 0.44549     0
16    ENCSR656JZL.HHEX.Hep-G2 Hep-G2              HHEX 0.44122     0
17   ENCSR800QIT.HNF1A.Hep-G2 Hep-G2             HNF1A 0.43966     0
18     ENCSR867WPH.REST.liver  liver              REST 0.43082     0
19    ENCSR669LCD.ATF4.Hep-G2 Hep-G2              ATF4 0.42999     0
20    ENCSR000BQX.NFIC.Hep-G2 Hep-G2              NFIC 0.42440     0
21     ENCSR837GTK.JUND.liver  liver              JUND 0.42418     0
22     ENCSR196HGZ.JUND.liver  liver              JUND 0.42393     0
23      ENCSR521IID.MAX.liver  liver               MAX 0.41846     0
24   ENCSR267DFA.FOXA1.Hep-G2 Hep-G2             FOXA1 0.41741     0
25 ENCSR887LYD.SMARCC2.Hep-G2 Hep-G2           SMARCC2 0.41697     0
26     ENCSR290ZOS.EGR1.liver  liver              EGR1 0.40709     0
27   ENCSR620YNB.KAT2B.Hep-G2 Hep-G2             KAT2B 0.40506     0
28   ENCSR000BRO.MYBL2.Hep-G2 Hep-G2             MYBL2 0.40453     0
29   ENCSR345YWJ.ZBTB33.liver  liver            ZBTB33 0.40031     0
30      ENCSR994YLZ.YY1.liver  liver               YY1 0.39534     0
31   ENCSR000BVM.NR2F2.Hep-G2 Hep-G2             NR2F2 0.39042     0
32    ENCSR444LIN.TCF7.Hep-G2 Hep-G2              TCF7 0.38687     0
33    ENCSR849FVL.GFI1.Hep-G2 Hep-G2              GFI1 0.38592     0
34   ENCSR029LBT.FOXP1.Hep-G2 Hep-G2             FOXP1 0.38317     0
35   ENCSR000EEW.ESRRA.Hep-G2 Hep-G2             ESRRA 0.38049     0
36    ENCSR000BHU.RXRA.Hep-G2 Hep-G2              RXRA 0.37795     0
37   ENCSR000EEU.HNF4A.Hep-G2 Hep-G2             HNF4A 0.37548     0
38   ENCSR791AGT.ZNF24.Hep-G2 Hep-G2             ZNF24 0.37417     0
39   ENCSR413AJG.FOXJ3.Hep-G2 Hep-G2             FOXJ3 0.37378     0
40     ENCSR000BJX.SP1.Hep-G2 Hep-G2               SP1 0.37328     0</code></pre>
</div>
</div>
<p>The above table shows that GSEA (in addition to TFEA.ChIP) also returns many liver-implicated transcriptional regulators. The above table is sorted in descending order of effect size (running enrichment score), and also displays the accession ID of the tissue sample that generated the ChIP seq data set. The TFs (transcription factors) can be in the table more than once.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>       TF  n
1    CTCF 31
2   EP300 11
3     MAX 11
4    JUND 10
5    REST 10
6   GABPA  9
7     MYC  7
8    TAF1  7
9  ZBTB33  7
10   ATF3  6</code></pre>
</div>
</div>
<p>The above table counts the number of line items for each TF; despite CTCF (a practically ubiquitous transcriptional and chromatin regulator) being so highly represented across data sets, its running effect size was not high enough to make it show up in the top 40 (5.8%) of line items. (Same goes for EP300.)</p>
<p>What this result shows is that the number of ChIP datasets under consideration in this package is not evenly distributed across all potential transcription factors, and so the number of times a transcription factor is identified as a hit is not a good measure of significance or salience.</p>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>